lib/graphql/schema/base_64_encoder.rb

```diff
@@ -13,6 +13,8 @@ def self.encode(unencoded_text, nonce: false)
       def self.decode(encoded_text, nonce: false)
         # urlsafe_decode64 is for forward compatibility
         Base64Bp.urlsafe_decode64(encoded_text)
+      rescue ArgumentError
+        raise GraphQL::ExecutionError, "Invalid cursor: #{encoded_text.inspect}"
```

Comment:
This code can also be used for base64-encoded IDs, how about changing the text a bit, eg

```suggestion
        raise GraphQL::ExecutionError, "Invalid input: #{encoded_text.inspect}"
```

?

Reply:
Makes total sense, done! 🙂